static/app/components/hovercard.tsx

```diff
@@ -40,45 +41,46 @@ interface HovercardProps extends Omit<UseHoverOverlayProps, 'isHoverable'> {
   tipColor?: ColorOrAlias;
 }
 
-function Hovercard({
+type UseOverOverlayState = ReturnType<typeof useHoverOverlay>;
+
+interface HovercardContentProps
+  extends Pick<
+    HovercardProps,
+    'bodyClassName' | 'className' | 'header' | 'body' | 'tipColor' | 'tipBorderColor'
+  > {
+  hoverOverlayState: Omit<UseOverOverlayState, 'isOpen' | 'wrapTrigger'>;
+}
+
+function useUpdateOverlayPositionOnContentChange({
+  update,
+}: Pick<UseOverOverlayState, 'update'>) {
+  const ref = useRef<HTMLDivElement | null>(null);
+
+  const onResize = useCallback(() => {
+    update?.();
+  }, [update]);
+
+  useResizeObserver({
+    ref,
+    onResize,
+  });
+
+  return ref;
```

Comment:
I think I would just inline the body of this function into `HovercardContent`. i think it actually makes it a bit more obvious what's going on with the ref then actually